app/proto/search.proto

```diff
@@ -198,10 +198,13 @@ message EventSearchReq {
 
   uint32 page_size = 10;
   string page_token = 11;
+  optional uint32 page_number = 18;
 }
 
 message EventSearchRes {
   repeated org.couchers.api.events.Event events = 1;
 
   string next_page_token = 2;
+  optional uint32 page_number = 3;
+  optional uint32 total_items = 4;
```

Comment:
Please remove all `optional`s.

So proto2 syntax had this required/optional thing. proto3 has everything be optional. It's a bit silly, optional/required was a good thing, but they simplified the language. In proto3 if something's missing it gets a default value (e.g. 0, empty string, etc, depending on type). proto2 was more powerful, you could define the default if it was missing.

Reply:
Thanks for the background! I've made the changes now and updated the tests.